cmake-proxies/cmake-modules/AudacityTesting.cmake

```diff
@@ -0,0 +1,143 @@
+      catch2/2.13.8
+      REQUIRED
+      ALWAYS_ALLOW_CONAN_FALLBACK
+   )
+
+   # Setup default CTest arguments when running from IDE
+
+   set(CMAKE_CTEST_ARGUMENTS "--output-on-failure;--verbose;${CMAKE_CTEST_ARGUMENTS}")
+
+   enable_testing()
+
+   #[[
+      add_unit_test(NAME name SOURCES file1 ... LIBRARIES lib1 ...)
+
+      Creates an executable called ${name}-test from the sources files ${file1}, ... and linked
+      to libraries ${lib1}, ... Catch2 is linked implicitly.
+
+      Create a CTest test called ${name}, that expects 0 code on success
+   ]]
+   function( add_unit_test )
+      cmake_parse_arguments(
+         ADD_UNIT_TEST # Prefix
+         "" # Options
+         "NAME" # One value keywords
+         "SOURCES;LIBRARIES"
+         ${ARGN}
+      )
+
+      if( NOT ADD_UNIT_TEST_NAME )
+         message( FATAL_ERROR "Missing required NAME parameter for the add_unit_test")
+      endif()
+
+      set( test_executable_name "${ADD_UNIT_TEST_NAME}-test" )
+
+      # Create test executable
+
+      add_executable( ${test_executable_name} ${ADD_UNIT_TEST_SOURCES} "${CMAKE_SOURCE_DIR}/tests/Catch2Main.cpp")
+      target_link_libraries( ${test_executable_name} ${ADD_UNIT_TEST_LIBRARIES} Catch2::Catch2 )
+
+      set_target_properties(
+         ${test_executable_name}
+         PROPERTIES
+            FOLDER "tests" # for IDE organization
+            RUNTIME_OUTPUT_DIRECTORY ${TESTS_DIR}
+            BUILD_RPATH ${_SHARED_PROXY_PATH}
+            # Allow running tests from Visual Studio by setting up the propper PATH
+            VS_DEBUGGER_ENVIRONMENT "PATH=${_SHARED_PROXY_PATH};%PATH%"
+      )
+
+      # Register unit test with CTest
+
+      add_test(
+         NAME
+            ${ADD_UNIT_TEST_NAME}
+         COMMAND
+            ${test_executable_name}
+      )
+
+      set_tests_properties(
+         ${ADD_UNIT_TEST_NAME}
+         PROPERTIES
+            LABELS "unit_tests"
+      )
+
+      if( WIN32 )
+         # On Windows, set the PATH so it points to the DLL location
+         # This is required to avoid invoking CopyLibs.
+
+         # CTest expects that ENVIRONMENT is a CMake list.
+         # Escape ';' so PATH is handled correctly
+         string(REPLACE ";" "\\;" escaped_path "$ENV{PATH}")
+
+         set_tests_properties(
+            ${ADD_UNIT_TEST_NAME}
+            PROPERTIES
+               ENVIRONMENT "PATH=$<SHELL_PATH:${_SHARED_PROXY_BASE_PATH}/$<CONFIG>>\\;${escaped_path}"
+         )
+      elseif( APPLE )
+         # We target an old version of macOS, disable std::uncaught_exceptions
+         target_compile_definitions( ${test_executable_name} PRIVATE CATCH_CONFIG_NO_CPP17_UNCAUGHT_EXCEPTIONS )
+         # Similar to Windows, but uses DYLD_FALLBACK_LIBRARY_PATH istead of PATH
+         set_tests_properties(
+            ${ADD_UNIT_TEST_NAME}
+            PROPERTIES
+               ENVIRONMENT "DYLD_FALLBACK_LIBRARY_PATH=$<SHELL_PATH:${_SHARED_PROXY_BASE_PATH}/$<CONFIG>>"
+         )
```

Comment:
I wonder if these fixes you figured out for dynamic libraries should be reused with image-compiler too.  Or have you done that already?

Maybe a common cmake function for such steps then.

Reply:
They can be reused, but the image-compiler is a) rarely built, b) does not test the functionality of a library. For these reasons, it is left as-is for now.